### PR TITLE
macos-trash 1.1.0 (new formula)

### PR DESCRIPTION
**Pull request by vladimyr**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

I'm not sure how to name this formula. The logical choice is just `trash` but that conflicts with existing one 🤔

### CODE REVIEW

**Comment by chenrui333** [comment]
The test might need to be updated 
```
==> /usr/local/Cellar/macos-trash/1.1.0/bin/trash testfile
“testfile” couldn’t be moved to the trash because you don’t have permission to access it.
Error: macos-trash: failed
```

**Comment by vladimyr** [comment]
> The test might need to be updated

Or completely removed 😞 Test sandbox does not allow accessing drive's trash folder so there is really nothing I can do about it. What's funny is that test is copied from [`trash-cli`](https://github.com/Homebrew/homebrew-core/blob/8653804/Formula/trash-cli.rb) and is probably broken there too.

That being said there is another problem with this one and [`dark-mode`](https://github.com/Homebrew/homebrew-core/blob/8653804/Formula/dark-mode.rb) and [`wallpaper`](https://github.com/Homebrew/homebrew-core/blob/8653804/Formula/wallpaper.rb) too. (Those are all Sindre's tools but I guess it is applicable to the vast majority of swift CLIs.) 

All three utilities have both:
1. required swift version that should be reflected through XCode's version in build dependency rule 
2. minimum macOS version required listed as `depends_on :macos` rule

but there is also a requirement to install swift runtime library if target os version is lower than `10.14.4` (note significant patch version): https://support.apple.com/kb/DL1998

Figuring 1. is quite easy. In this case 	`Package.swift` says it requires swift 5.1 that got shipped with XCode 11.0 (https://swift.org/download/#releases). But what to do in 2. case? I can't really put `:mojave` there due to the required patch version. Also, I shouldn't put a restriction on a tool that works all down to `:yosemite` by setting `:catalina` constraint just to avoid installing swift runtime? 🤔 

Any ideas?

**Comment by vladimyr** [comment]
I'm not a swift expert but I'm guessing that what I just described is the reason for following https://github.com/mxcl/homebrew-made/blob/b687f14/swift-sh.rb#L13-L33 and compare that with the current formula hosted here https://github.com/vladimyr/homebrew-core/blob/8ad1682/Formula/swift-sh.rb

Looks like we have a bigger problem 🤷

**Comment by chenrui333** [comment]
Looks like the build got stuck, let's rebase and give another try. :)

**Comment by SMillerDev** [comment]
Also stuck

**Comment by vladimyr** [comment]
> Looks like the build got stuck, let's rebase and give another try. :)

> Also stuck

Built 🚀

File: Formula/macos-trash.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  desc "Move files and folders to the trash"
+  homepage "https://github.com/sindresorhus/macos-trash"
+  url "https://github.com/sindresorhus/macos-trash/archive/1.1.0.tar.gz"
+  sha256 "31c09d385bb50b0f76818a1fe2c850cf56b9575c9fa27ea963cba38dfaba7d04"
+  head "https://github.com/sindresorhus/macos-trash.git"
+
+  depends_on :xcode => ["11.0", :build]
+  depends_on :macos => :yosemite
+
+  conflicts_with "trash", :because => "both install a `trash` binary"
```
Comment (dawidd6) [inline Formula/macos-trash.rb:11]: I think that we will need to update `trash` and `trash-cli` formulae, to add a new conflict with this formula there too.

Comment (SMillerDev) [reply Formula/macos-trash.rb:11]: It should be in this one IYAM. Otherwise there's a moment where it's conflicting without people knowing.

Comment (vladimyr) [reply Formula/macos-trash.rb:11]: > I think that we will need to update `trash` and `trash-cli` formulae, to add a new conflict with this formula there too. That can be done in separate PRs though.

Done ✅

**Review by dawidd6 — APPROVED** [review]

File: Formula/macos-trash.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class MacosTrash < Formula
+  desc "Move files and folders to the trash"
+  homepage "https://github.com/sindresorhus/macos-trash"
+  url "https://github.com/sindresorhus/macos-trash/archive/1.1.0.tar.gz"
+  sha256 "31c09d385bb50b0f76818a1fe2c850cf56b9575c9fa27ea963cba38dfaba7d04"
+  head "https://github.com/sindresorhus/macos-trash.git"
+
+  depends_on :xcode => ["11.0", :build]
```
Comment (chenrui333) [inline Formula/macos-trash.rb:8]: Retrospectively looking, I think your `xcode 11.0` already implied this has to be built with Mojave and later, not quite sure why we sill need `:macos => :yosemite`.

Also, it looks like this is the only formula using `:macos => :yosemite` :)

**Pull request closed without merging**